File: tests/fixtures/supabase/supabase-fail-fast-env-validation-fixed/02-non-env-args-adversarial.ts
import { createClient } from '@supabase/supabase-js';

// Adversarial: this calls createClient with no apparent presence check,
// which could look like the same bug at a glance — but neither argument is
// derived from process.env at all (they're plain function parameters, e.g.
// a test helper that wires in fixed credentials), so there's nothing to
// fail-fast validate and this must not be flagged.
export function createTestClient(url: string, key: string) {
  return createClient(url, key);
}

File: tests/fixtures/supabase/supabase-fail-fast-env-validation-fixed/02_non_env_args_adversarial.py
from supabase import create_client

# Not sourced from os.environ at all — out of scope for this rule regardless
# of whether the literal is "valid".
supabase = create_client("https://example.supabase.co", "public-anon-key")


def make_test_client(url: str, key: str):
    # Plain function params, not env-derived — should not be flagged either.
    return create_client(url, key)
